Fetch/Fetch.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  Fetch.swift
+//  Fetch
+//
+//  Created by David Hardiman on 13/02/2016.
+//  Copyright © 2016 David Hardiman. All rights reserved.
+//
+
+import Foundation
+
+/**
+ The result of a fetch request
+ 
+ - Successful: The fetch and parse was successful
+ - Failure:    The fetch returned, but the object returned represents a failure
+ - Error:      The fetch failed with an error
+ */
+public enum FetchResult<T: Parsable> {
+    case Successful(T)
+    case Failure(T)
```

Comment:
Maybe this case should also return an `ErrorType` that `Parsable` is responsible for generating